src/main/java/seedu/address/model/inventorymodel/InventoryBook.java

```diff
@@ -117,6 +118,11 @@ public void commit() {
         if (!inventoryBookStateList.isEmpty()) {
             inventoryBookStateList = inventoryBookStateList.subList(0, inventoryBookStatePointer + 1);
         }
+        if (inventoryBookStateIsFull()) {
+            inventoryBookStateList.remove(0);
+            inventoryBookStatePointer--;
+        }
+
```

Comment:
In my experience with CommandHistory's previousCommand(), be wary of the corner case if suppose:
you have 2 states, state#1, state#2

undo() should return you to state#2 and not state#1.
Could you give that a shot? Maybe either via junit tests or manual testing

Reply:
This is a complication from having the pointer pointing at the last element of the list

Reply:
Nvm ignore me i've read through your implementation and realised its different

Reply:
Manual testing shows that it works as expected when the limit is 1 or 2

Reply:
Thanks!!